### ISSUE TITLE
SecurityWrapperResponse setHeader error message is unclear

### ISSUE THREAD

**Issue by ThrawnCA** [ESAPI/esapi-java-legacy#536]
When `SecurityWrapperResponse.setHeader` reports a validation failure, the message doesn't indicate whether it's the header name or header value that's failing.

**Comment by kwwall** [comment]
@ThrawnCA  - Are you referring to the SECURITY_FAILURE message that is logged? There probably was a (wrong) assumption made that that information was included in the exception message itself. Will take a look.

**Comment by ThrawnCA** [comment]
That sounds right. Just need to know whether it's the name or the value that's failing.

(Would also be nice to see at a glance whether it's failing the regex or the length limit, but c'est la vie. In our recent case, the length limit within the regex was shorter than the length limit imposed by the configuration, which was as confusing as you'd think.)

**Comment by kwwall** [comment]
@ThrawnCA - Alright, I'll see what I can do and maybe ask you to look at the PR before we merge it to the 'develop' branch.

**Comment by jeremiahjstacey** [comment]
addHeader had the same behavior as well.  Please see https://github.com/ESAPI/esapi-java-legacy/pull/537

Both methods have been updated similarly to:
```
  SecurityConfiguration sc = ESAPI.securityConfiguration();
        String strippedName = StringUtilities.stripControls(name);
        String strippedValue = StringUtilities.stripControls(value);
        String safeName = null;
        String safeValue = null;
        try {
            safeName = ESAPI.validator().getValidInput("addHeader", strippedName, "HTTPHeaderName", sc.getIntProp("HttpUtilities.MaxHeaderNameSize"), false);
        } catch (ValidationException e) {
            logger.warning(Logger.SECURITY_FAILURE, "Attempt to add invalid header denied [HEADER_NAME]", e);
        }
        
        try {
            safeValue = ESAPI.validator().getValidInput("addHeader", strippedValue, "HTTPHeaderValue", sc.getIntProp("HttpUtilities.MaxHeaderValueSize"), false);
        } catch (ValidationException e) {
            logger.warning(Logger.SECURITY_FAILURE, "Attempt to add invalid header denied [HEADER_VALUE]", e);
        }
```

**Comment by kwwall** [comment]
Does anyone think that we should include the header **_name_** in these log messages as well?

I could see a case in a HTTP proxy or servlet filter where someone just loops on all the response headers that are being returned and then adding them to a new `HttpServletRequest` object that is being passed back and then some, but not all of the header checks fail as they are added so the ESAPI user ends up with 2 or 3 log messages all that say:
&nbsp;&nbsp;&nbsp;&nbsp;`Attempt to add invalid header denied [HEADER_NAME]`
out of a total of maybe a half dozen or so response headers, but the ESAPI user can't tell from the log messages which header names failed for that given HTTP response.

I know that isn't exactly what @ThrawnCA requested but I could potentially see that becoming an issue at some point, so if anyone thinks that may be a problem and it would be helpful to show the HTTP response header name (but never the value), maybe we ought to do this while we are thinking about it. I'll give everyone until Sunday evening to think about it. If the rest of you think it is good enough as PR #537 now stands, I will go ahead and merge it. If anyone things the header name should be included, I will update the description of this GitHub issue with that detail and then add a comment to the PR so that @jeremiahjstacey can add that (assuming he has no objections).

I'm personally okay either way, since I'm more of an ESAPI provider than an ESAPI user, but try to think past the ordinary use cases so we don't have to come back an revisit this a few years later. Thanks.

**Comment by jeremiahjstacey** [comment]
Adding the Header Name would not be too difficult, and having more identifiable information in the logs is always handy when diagnostics are required.  I am very much in favor of usable & meaningful logs.

Given the snippet I've offered above, I would like to clarify whether we're talking about adding  the original method parameter `name` (1)  or the trimmed and validated `safeName` value (2)?   

1. Using the `name`  parameter would be a simple addition to the logging lines.  
2. Using the `safeName` reference will require additional logic in the catch blocks to ensure the field is not nulll.

**Comment by kwwall** [comment]
@jeremiahjstacey - I'm for simple. I think the original 'name' parameter is fine. If it turns out that it is not "safe" in itself, we will let ESAPI's "safe logging" take care of that.

**Comment by jeremiahjstacey** [comment]
Updates applied to PR.

**Comment by kwwall** [comment]
Closed via @jeremiahjstacey PR #537
